app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -79,4 +139,51 @@ class KiwixMainActivity : CoreMainActivity() {
       it.onNewIntent(intent, this)
     }
   }
+
+  override fun onNavigationItemSelected(item: MenuItem): Boolean {
+    when (item.itemId) {
+      R.id.menu_support_kiwix -> openSupportKiwixExternalLink()
+      R.id.menu_settings -> openSettingsActivity()
+      R.id.menu_help -> start<HelpActivity>()
+      R.id.menu_host_books -> start<ZimHostActivity>()
+      R.id.menu_history -> openHistoryActivity()
+      R.id.menu_bookmarks_list -> openBookmarksActivity()
+      else -> return false
+    }
+    return true
+  }
+
+  private fun openSupportKiwixExternalLink() {
+    val intent = Intent(
+      Intent.ACTION_VIEW,
+      Uri.parse("https://www.kiwix.org/support")
+    ).putExtra(EXTRA_EXTERNAL_LINK, true)
+    alertDialogShower.show(
+      KiwixDialog.ExternalLinkPopup,
+      { startActivity(intent) }, {
+        sharedPreferenceUtil.putPrefExternalLinkPopup(false)
+        startActivity(intent)
+      }
+    )
+  }
+
+  private fun openSettingsActivity() {
+    startActivityForResult(
+      Intents.internal(CoreSettingsActivity::class.java),
+      REQUEST_PREFERENCES
+    )
+  }
+
+  private fun openHistoryActivity() {
+    startActivityForResult(
+      intent<HistoryActivity>(),
+      REQUEST_HISTORY_ITEM_CHOSEN
+    )
+  }
+
+  private fun openBookmarksActivity() {
+    startActivity(
+      intent<BookmarksActivity>()
```

Comment:
This doesn't do what CoreReaderFragment does 

```
 boolean goToBookmarks() {
    saveTabStates();
    Intent intentBookmarks = new Intent(getActivity(), BookmarksActivity.class);
    startActivityForResult(intentBookmarks, BOOKMARK_CHOSEN_REQUEST);
    return true;
  }
```

Reply:
That is, the tab state is not saved. Any nice way of making the fragment save the tab state from the navigation activity?

Reply:
```

private fun openBookmarksActivity() {
    supportFragmentManager.fragments.filterIsInstance<ReaderFragment>()
      .forEach(ReaderFragment::saveTabStates)
    startActivityForResult(intent<BookmarksActivity>(), BOOKMARK_CHOSEN_REQUEST)
  }
```

Reply:
I can't remember the tabState saving flow, if it just happened `onPause` then we probably wouldn't have to worry overmuch about actually invoking it

Reply:
also if I remember the intent extension function it accepts a receiver function do you can do `intent<Activity>{ putExtra() }`

Reply:
hmm, maybe startActivityForResult was the problem. Just noticed that the tab states are saved onPause and it did not work before setting startActivityForResult.